test/suggestion.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2021 Maneesh P M <manu.pm55@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * is provided AS IS, WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, and
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#include <zim/zim.h>
+#include <zim/archive.h>
+#include <zim/search.h>
+#include <zim/writer/creator.h>
+#include <zim/writer/item.h>
+#include <zim/writer/contentProvider.h>
+
+#include "tools.h"
+
+#include "gtest/gtest.h"
+
+namespace {
+
+  class TestItem : public zim::writer::BasicItem {
+      public:
+      TestItem(const std::string& path, const std::string& mimetype, const std::string& title)
+        : BasicItem(path, mimetype, title) {}
+      
+      virtual std::unique_ptr<zim::writer::ContentProvider> getContentProvider() const {
+        return std::unique_ptr<zim::writer::ContentProvider>(new zim::writer::StringProvider(""));
+      }
+  };
+
+  std::string createTestZim(std::vector<std::string> titles) {
+    zim::unittests::TempFile temp("testZim");
+    auto tempPath = temp.path() + ".zim";
+    
+    zim::writer::Creator creator;
+    creator.configIndexing(true, "eng");
+    creator.startZimCreation(tempPath);
+
+    // add dummy items with given titles
+    for (auto title : titles) {
+      std::string path = "dummyPath" + title;
+      auto item = std::make_shared<TestItem>(path, "plain/text", title);
+      creator.addItem(item);
+    }
+
+    creator.addMetadata("Title", "This is a title");
+    creator.setMainPath("foo");
```

Comment:
The line doesn't make sense, since there is no such entry in this ZIM archive.

Reply:
@maneeshpm Probabky we should set the mainpage to the first page in the list,

Reply:
The mainPath isn't playing any role in our tests anyway. Removing it.